### PR TITLE
Refactor: Strong types for UID UIDValidity and IMAP Sequence

### PR DESCRIPTION
**Pull request by LBeernaertProton**

### CODE REVIEW

File: imap/seqset.go [hunk 1]
```diff
@@ -26,6 +27,12 @@ func (seqval SeqVal) String() string {
 
 type SeqSet []SeqVal
```
Comment (cuthix) [inline imap/seqset.go:28]: SeqVal has two `int`s should we use `uint32` ?

File: internal/response/exists.go [hunk 2]
```diff
@@ -5,14 +5,14 @@ import (
...
-	count int
+	count uint32
 }
 
 func Exists() *exists {
 	return &exists{}
 }
 
-func (r *exists) WithCount(n int) *exists {
+func (r *exists) WithCount(n uint32) *exists {
```
Comment (jameshoulahan) [inline internal/response/exists.go:15]: Technically this is also of type `imap.Seq` -- an exists response returns the sequence number of the last message in the mailbox

File: internal/backend/user.go [hunk 3]
```diff
@@ -29,7 +29,7 @@ type user struct {
 
 	db *db.DB
 
-	states      map[int]*state.State
+	states      map[state.StateID]*state.State
 	statesLock  sync.RWMutex
 	nextStateID int
```
Comment (jameshoulahan) [inline internal/backend/user.go:34]: why not make this of type `state.StateID`?

File: imap/seqset.go (hunk 1 above)
Comment (LBeernaertProton) [reply imap/seqset.go:28]: Good catch. fixed.

File: internal/response/exists.go (hunk 2 above)
Comment (LBeernaertProton) [reply internal/response/exists.go]: Fixed.

File: internal/backend/user.go (hunk 3 above)
Comment (LBeernaertProton) [reply internal/backend/user.go]: Good idea. Done.

**Review by jameshoulahan — APPROVED** [review]
Wondering whether the state's ID should be private and only accessible via a getter, just to help enforce the user/state/backend separation we're trying to create. But I don't mind how it is now -- approved

**Pull request merged**